client/src/js/services/util.js

```diff
@@ -21,15 +22,16 @@ function UtilService($filter, moment) {
       if (!isAngularAttribute) {
 
         // Only format and assign dirty values that have changed
-        if (value.$dirty) {
-
-          // any standard ng-model element will provide an $modelValue, according
-          // to the latest 2.x standards more complex bhima components or bhima
-          // component wrappers will expose $bhValue
-
-          // accounts for empty string values
-          response[key] = angular.isDefined(value.$modelValue) ? value.$modelValue : value.$bhValue;
+        if (requireDirty && !value.$dirty) {
+          return;
         }
+        
+        // any standard ng-model element will provide an $modelValue, according
+        // to the latest 2.x standards more complex bhima components or bhima
+        // component wrappers will expose $bhValue
```

Comment:
The $bhValue should probably be well documented. :)  I had forgotten about this myself.